apps/fabric-website/src/components/IconGrid/IconGrid.tsx

```diff
@@ -56,36 +58,88 @@ export class IconGrid extends React.Component<IIconGridProps, IIconGridState> {
     );
   }
 
+  public async componentDidMount() {
+    // Load the icons async after mount to avoid them being bundled and parsed with the page,
+    // due to their large bundle size.
+    switch (this.props.iconType) {
+      case 'font':
+        // The explicit json-loader! import helps prevent this from being processed by TS
+        // (we want only webpack to handle it)
+        const fontIcons = (await import('json-loader!@uifabric/icons/lib/data/AllIconNames.json')) as IIconInfo[];
+        // There's also a metadata entry in this file which doesn't have a name
+        this.setState({ icons: fontIcons.filter(icon => !!icon.name) });
+        break;
+
+      case 'core':
+        const coreIcons = (await import('json-loader!office-ui-fabric-core/src/data/icons.json')) as IIconInfo[];
+        this.setState({ icons: coreIcons });
+        break;
+
+      case 'svg':
+        const MDL2Icons = await import('@fluentui/react-icons');
+        this.setState({
+          icons: Object.keys(MDL2Icons)
+            .map(key => {
+              const IconComponent = MDL2Icons[key];
+              if (
+                typeof IconComponent === 'function' &&
+                key !== 'createSvgIcon' &&
+                String(IconComponent).indexOf('return React.createElement') !== -1
```

Comment:
This check won't work once the files are bundled with webpack ("React" gets called something entirely different). So probably the next best thing is to check whether the name looks like an icon component: `/^[A-Z].*Icon$/`, starts with uppercase and ends with Icon.